google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/SpannerInterceptorProvider.java

```diff
@@ -16,28 +16,47 @@
 package com.google.cloud.spanner.spi.v1;
 
 import com.google.api.gax.grpc.GrpcInterceptorProvider;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
 import io.grpc.ClientInterceptor;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
- * For internal use only. 
- * An interceptor provider that provides a list of grpc interceptors for {@code GapicSpannerRpc}
- * to handle logging and error augmentation by intercepting grpc calls.
+ * For internal use only. An interceptor provider that provides a list of grpc interceptors for
+ * {@code GapicSpannerRpc} to handle logging and error augmentation by intercepting grpc calls.
  */
-class SpannerInterceptorProvider implements GrpcInterceptorProvider {
+@VisibleForTesting
+public class SpannerInterceptorProvider implements GrpcInterceptorProvider {
 
-  private static final List<ClientInterceptor> clientInterceptors =
+  private static final List<ClientInterceptor> defaultInterceptors =
       ImmutableList.of(
           new SpannerErrorInterceptor(),
-          new LoggingInterceptor(Logger.getLogger(GrpcSpannerRpc.class.getName()), Level.FINER),
+          new LoggingInterceptor(Logger.getLogger(GapicSpannerRpc.class.getName()), Level.FINER),
           WatchdogInterceptor.newDefaultWatchdogInterceptor());
 
+  private final List<ClientInterceptor> clientInterceptors;
+
+  private SpannerInterceptorProvider(List<ClientInterceptor> clientInterceptors) {
+    this.clientInterceptors = clientInterceptors;
+  }
+
+  public static SpannerInterceptorProvider createDefault() {
+    return new SpannerInterceptorProvider(defaultInterceptors);
+  }
+
+  public SpannerInterceptorProvider with(ClientInterceptor clientInterceptor) {
```

Comment:
Are we actually using this method apart from testing? If not, is it possible to make this package-private?

Reply:
It's used here: https://github.com/GoogleCloudPlatform/google-cloud-java/pull/3362/files#diff-41c3438699a47c682e96c1e19d56fb9dR60. It's for integration testing only, but the test is in a different package.

Reply:
In that case, add `@VisibleForTesting`

Reply:
Please don't use `@VisibleForTesting` - that exposes Guava. Use `@InternalApi("Exposed for testing")` instead.